helix-term/src/commands/typed.rs

```diff
@@ -352,13 +352,26 @@ fn write_impl(
     };
 
     if fmt.is_none() {
+        if config.insert_final_newline {
+            insert_final_newline(doc, view);
+        }
```

Comment:
I discussed this again with @the-mikedavis. I think we actually don't need this special case here. Not all formatters insert a trailing newlino (for those the new option would not work with the current implementation). Those that do won't have a problem as long as we do it *before* sending a request to the formatter/LSP (if you do it afterward you will endup with the formatter not working anymore).


The call to `insert_final_newline`  should be moved before the creation of the `fmt` callback above.

Reply:
Makes sense; done!